benchmarks/benchmarks/spatial.py

```diff
@@ -273,6 +273,29 @@ def time_count_neighbors_shallow(self, mn1n2, Nr):
         """
         self.T1s.count_neighbors(self.T2s, self.r)
 
+
+class Remove(Benchmark):
+    params = [
+        [
+          (2, 20000),
+          (8, 20000),
+          (16, 20000)
+        ],
+        BOX_SIZES, LEAF_SIZES
+    ]
+
+    def setup(self, mn, boxsize, leafsize):
+        m, n = mn
+        dataset = np.random.uniform(size=(n, m))
+        self.T = cKDTree(dataset, boxsize=boxsize, leafsize=leafsize)
+
+    def time_remove(self, mn, boxsize, leafsize):
+        """
+        Time to remove one point from cKDTree.
+        """
+        self.T.remove(0)
```

Comment:
I run this benchmark and the results were I would say "suspiciously" good, on the order of 300 ns or only a few python function calls. After investigating, the `setup` method isn't re-run between iterations of the timing loop, so the point is only actually removed from the tree once.

I'm not sure if there's a good way to run non-timed code (@pv?) but you really need to build a new tree for each call to `time_remove`:
```python
tree = cKDTree(self.dataset, boxsize=boxsize, leafsize=leafsize)
tree.remove(0)
```

But then the timing is dominated by the time to build the tree, to the point where `remove` is lost in the measurement error. Calling `remove` in a loop can bring it above the noise floor though.

Reply:
Using that technique I estimate this is somewhere on the order of 100x faster than building a new tree from scratch. Somewhere in the 10s of microseconds.

Reply:
It seems it can be done properly with pytest-benchmark (https://stackoverflow.com/questions/51288551/pytest-benchmark-run-setup-on-each-benchmark-iteration)

I've thought to initialize a list of cKDTrees in the setup method. In this case, time_remove has a little overhead due to an index incrementation, but I don't know the number of iterations